### ISSUE TITLE
open /dev/null: too many open files

### ISSUE THREAD

**Issue by dashaus** [tockins/realize#5]
I get the error "open /dev/null: too many open files". I don't know what is the cause. In this moment, I'm using github.com/pilu/fresh and not happen this.

**Comment by dashaus** [comment]
The error is raised in build.Run() from GoInstall(), GoFmt() and GoBuild() methods

**Comment by Asoseil** [comment]
how many files do you have? could be a problem of the fsnotify package.

**Comment by conventi** [comment]
@dashaus thanks for the tip, we'll investigate as soon as possible how fresh work.

**Comment by conventi** [comment]
Probably it is a problem related to operating system limits as you can read in the last [FAQ](https://github.com/fsnotify/fsnotify/wiki/FAQ) of fsnotify. @dashaus have you ran realize and fresh at same time? In this case you may have reached the maximum limit of files that can be watched.

**Comment by dashaus** [comment]
@Asoseil I don't know. How can I know it?
@conventi No, only realize. There is any possibility to know when files is trying to open `realize`?

**Comment by Asoseil** [comment]
In your config yaml, under the Watcher obj, add "preview:true"

**Comment by dashaus** [comment]
I changed to "github.com/howeyc/fsnotify" package (the same that uses fresh), then I execute the command go install in the realize package and try it but the result is the same...

**Comment by conventi** [comment]
@dashaus if you are on a MAC OS or a GNU/Linux system you can count the number of files (recursively) by execute `find . -type f | wc -l` form working directory. Let us know!

**Comment by dashaus** [comment]
@conventi I'am on Mac OS X. Sorry by not saying it before. The total is 3092 files.

**Comment by conventi** [comment]
@dashaus you have 3093 files that you want to watch (have you cd into right directory)? Sorry if I ask but we have to retrive all informations to debug realize. Thanks for help!

**Comment by dashaus** [comment]
Yes. Is the total files that I want to watch.

**Comment by dashaus** [comment]
```bash
iMac-de-Emilio:monoculum emilio$ ulimit -a
core file size          (blocks, -c) 0
data seg size           (kbytes, -d) unlimited
file size               (blocks, -f) unlimited
max locked memory       (kbytes, -l) unlimited
max memory size         (kbytes, -m) unlimited
open files                      (-n) 8192
pipe size            (512 bytes, -p) 1
stack size              (kbytes, -s) 8192
cpu time               (seconds, -t) unlimited
max user processes              (-u) 709
virtual memory          (kbytes, -v) unlimited
```

**Comment by conventi** [comment]
Ok thanks a lot! At this point this is obviously a bug related to realize and we will investigate into _fresh package_ for resolve this issue.

**Comment by dashaus** [comment]
OK, thanks you!

**Comment by dashaus** [comment]
Ok. I found a solution. In the function routines(), I added to the end `wg.Wait()`, and now it works.

```go
routines := func() {
		channel = make(chan bool)
		err = p.fmt()
		if err == nil {
			wr.Add(1)
			go p.build()
			go p.install(channel, &wr)
			wg.Wait() // I ADDED THIS LINE AND NOW WORKS!
		}
	}
```

**Comment by dashaus** [comment]
Also, inside the routines function, when an error is raised, this error is not displayed in the terminal.

**Comment by conventi** [comment]
@dashaus thanks a lot for your debug. We will testing your fix and if it's ok for all our cases we will integrate it!

**Comment by dashaus** [comment]
@conventi You are welcome. Thank you for the package. I was long time looking for a substitute for `fresh`!

**Comment by Asoseil** [comment]
Sorry, i lost your contribute with the last merge. added back just now. do you have still problem with a huge files number?

**Comment by dashaus** [comment]
The last commit does that live reload not works... Is by this that I closed my pull request #6
